### PR TITLE
Use correct 1-1 room avatar after users leave

### PR DESCRIPTION
**Pull request by lukebarnard1**
The correct 1-1 avatar is used with rooms in which there are only two users with either "join" or "invite" as their membership (importantly, not "leave" or otherwise).

(This is important when a user moves accounts and re-joins previously left 1-1 chats)

### CODE REVIEW

**Comment by ara4n** [comment]
will this not have the side-effect that if i'm in a 1:1 with someone and they leave, the icon will change to just be me, rather than showing them?  This will be quite unintuitive, given the history is with them...

**Comment by ara4n** [comment]
lgtm, assuming this matches the logic for calculating the name. (it feels wrong that these are maintained entirely separately........)

**Comment by lukebarnard1** [comment]
It would be nice to push this sort of stuff back up to the js-sdk somehow. 

I'm now thinking that `leftUserIds` should probably only contain users that actually left... If there's one person in the room and one person got _kicked_, they should probably not be the room avatar?... The spec says (and this makes a lot of sense) that you should only " consider m.room.member events for users other than the logged-in user, with membership: join or membership: invite. i.e. other members should be totally ignored. 

TL;DR this PR currently ignores the room name spec.

**Comment by lukebarnard1** [comment]
Closing as this is an obscure edge-case of swapping accounts

**Comment by lukebarnard1** [comment]
FTR @dbkr looked this over and gave the go-ahead.
